argoverse/evaluation/stereo/utils.py

```diff
@@ -0,0 +1,160 @@
+# <Copyright 2021, Argo AI, LLC. Released under the MIT license.>
+"""Stereo utilities to support the Argoverse stereo evaluation."""
+
+from pathlib import Path
+from typing import List
+
+import cv2
+import disparity_interpolation
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+
+from argoverse.evaluation.stereo.constants import (
+    DEFAULT_ABS_ERROR_THRESHOLDS,
+    DEFAULT_REL_ERROR_THRESHOLDS,
+    LOG_COLORMAP,
+)
+
+
+def compute_disparity_error(
+    pred_fpath: Path,
+    gt_fpath: Path,
+    gt_obj_fpath: Path,
+    figs_fpath: Path,
+    abs_error_thresholds: List[int] = DEFAULT_ABS_ERROR_THRESHOLDS,
+    rel_error_thresholds: List[float] = DEFAULT_REL_ERROR_THRESHOLDS,
+    save_disparity_error_image: bool = False,
+) -> pd.DataFrame:
+    """Compute the disparity error metrics."""
+    pred_disparity = cv2.imread(str(pred_fpath), cv2.IMREAD_ANYCOLOR | cv2.IMREAD_ANYDEPTH)
+    gt_disparity = cv2.imread(str(gt_fpath), cv2.IMREAD_ANYCOLOR | cv2.IMREAD_ANYDEPTH)
+    gt_obj_disparity = cv2.imread(str(gt_obj_fpath), cv2.IMREAD_ANYCOLOR | cv2.IMREAD_ANYDEPTH)
+
+    pred_disparity = np.float32(pred_disparity) / 256.0
```

Comment:
are these stored as uint8? i thought we were saving them as uint16

Reply:
These are inverse depth (disparity in pixel units), so uint8 is enough. We can always recover true depth with great precision.

Reply:
Fixing my previous comment: the disparity maps are saved as uint16 PNG images. The true disparity value for each pixel is computed by converting the uint16 value to float and dividing it by 256.

Reply:
Cool. Do you mind adding as an inline comment here that we are reading uint16 files?

What is the resolution and min/max values we can represent without uint16/256 representation? maybe useful to document somewhere